### ISSUE TITLE
[FEATURE] Support inputlookup Command in PPL. 

### ISSUE THREAD

**Issue by LantaoJin** [opensearch-project/opensearch-spark#638]

**Comment by penghuo** [comment]
+1 on (Preferred) Upload CSV files to external URL.
One concern is how to prevent users from accessing any data in the local filesystem, as this poses a security risk.

**Comment by YANG-DB** [comment]
I agree with @penghuo this is a possible security concern - I would propose using a different approach:
use the dashboard for loading a csv file into and index and using this index for the lookup

**Comment by brijos** [comment]
I hate to be that guy, but I know of those in the community who would want to load the CSV into their index as well as those who want to load the CSV into cloud storage. From a priority perspective, index should be the first as it is the easiest (assuming the analyst has write access to the cluster). Dealing with cloud storage introduces permissions friction.

**Comment by LantaoJin** [comment]
> +1 on (Preferred) Upload CSV files to external URL. One concern is how to prevent users from accessing any data in the local filesystem, as this poses a security risk.

A straightforward solution is allowing the `s3://` schema URL only in product.

> From a priority perspective, index should be the first as it is the easiest

Yes. I got the priorities. We have the `lookup` issue https://github.com/opensearch-project/opensearch-spark/issues/620 opened. This issue is for the requirement of loading data from a CSV (similar to the the `inputlookup` command in Splunk).
